sql/core/src/test/scala/org/apache/spark/sql/TPCDSQueryTestSuite.scala

```diff
@@ -149,7 +187,12 @@ class TPCDSQueryTestSuite extends QueryTest with TPCDSBase with SQLQueryTestHelp
         classLoader = Thread.currentThread().getContextClassLoader)
       test(s"$name-v2.7") {
         val goldenFile = new File(s"$baseResourcePath/v2_7", s"$name.sql.out")
-        runQuery(queryString, goldenFile)
+        val (schema, output) = runQuery(queryString, goldenFile)
+        if (!regenerateGoldenFiles) {
+          shuffleConfSet.foreach { testConf =>
+            runQueryWithShuffleConf(queryString, schema, output, testConf)
```

Comment:
we can change the `runQuery` to `runQuery(query: String, goldenFile: File, needSort: Boolean)` and reuse it to replace the `runQueryWithShuffle`. I don't think reading the output file multiple time is a problem

Reply:
OK